glTFLoader/Interface.cs

```diff
@@ -175,39 +145,75 @@ private static void ReadBinaryHeader(BinaryReader binaryReader)
+                var bufferFilePath = Path.Combine(Path.GetDirectoryName(gltfFilePath), asset);
+                return File.ReadAllBytes(bufferFilePath);
+            };
+        }
+
+        public static Byte[] LoadBinaryBuffer(this Gltf model, string gltfFilePath, int bufferIndex)
+        {
+            return LoadBinaryBuffer(model, GetExternalFileSolver(gltfFilePath), bufferIndex);            
+        }
+
         public static Stream OpenImageFile(this Gltf model, string gltfFilePath, int imageIndex)
         {
+            return OpenImageFile(model, GetExternalFileSolver(gltfFilePath), imageIndex);
+        }
+
+        public static Byte[] LoadBinaryBuffer(this Gltf model, Func<string,Byte[]> externalReferenceSolver, int bufferIndex)
+        {
+            var buffer = model.Buffers[bufferIndex];
+
+            if (buffer.Uri == null) return externalReferenceSolver(null);
+
+            if (buffer.Uri.StartsWith(EMBEDDEDOCTETSTREAM))
+            {
+                var content = buffer.Uri.Substring(EMBEDDEDOCTETSTREAM.Length);
+                return Convert.FromBase64String(content);
+            }
+
+            return externalReferenceSolver(buffer.Uri);
+        }
+
+        public static Stream OpenImageFile(this Gltf model, Func<string, Byte[]> externalReferenceSolver, int imageIndex)
+        {
             var image = model.Images[imageIndex];
 
             if (image.BufferView.HasValue)
             {
                 var bufferView = model.BufferViews[image.BufferView.Value];
 
-                var bufferBytes = model.LoadBinaryBuffer(gltfFilePath, bufferView.Buffer);
+                var bufferBytes = model.LoadBinaryBuffer(externalReferenceSolver, bufferView.Buffer);
 
                 return new MemoryStream(bufferBytes, bufferView.ByteOffset, bufferView.ByteLength);
             }
-            
-            if (image.Uri.StartsWith("data:image/"))
-            {
-                string content = null;
 
-                if (image.Uri.StartsWith(EMBEDDEDPNG)) content = image.Uri.Substring(EMBEDDEDPNG.Length);
-                if (image.Uri.StartsWith(EMBEDDEDBMP)) content = image.Uri.Substring(EMBEDDEDBMP.Length);
-                if (image.Uri.StartsWith(EMBEDDEDGIF)) content = image.Uri.Substring(EMBEDDEDGIF.Length);
-                if (image.Uri.StartsWith(EMBEDDEDJPEG)) content = image.Uri.Substring(EMBEDDEDJPEG.Length);
-                if (image.Uri.StartsWith(EMBEDDEDTIFF)) content = image.Uri.Substring(EMBEDDEDTIFF.Length);                
+            if (image.Uri.StartsWith("data:image/")) return OpenEmbeddedImage(image);
+
+            var imageData = externalReferenceSolver(image.Uri);
+
+            return new MemoryStream(imageData);
+        }
+
+        private static Stream OpenEmbeddedImage(Image image)
+        {
+            string content = null;
+
+            if (image.Uri.StartsWith(EMBEDDEDPNG)) content = image.Uri.Substring(EMBEDDEDPNG.Length);

```

Comment:
You still have more types than just PNG and JPG. Remove BMP, GIF, and TIFF here and the constants.

Reply:
You mean, only PNG and JPG are supported?

Reply:
Yes.

https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#imageuri